administrator/components/com_associations/helpers/associations.php

```diff
@@ -260,4 +260,99 @@ public static function getComponentProperties($key = '')
 
 		return $cp[$key];
 	}
+
+	/**
+	 * Check if user is allowed to edit own item
+	 *
+	 * @param   string  $componentKey  The component properties.
+	 * @param   JTable  $item          Database row from the component.
+	 *
+	 * @return  boolean.
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public static function allowEditOwn($componentKey = '', $item = null)
+	{
+		$user  = JFactory::getUser();
+
+		if (isset($item->{$componentKey->fields->created_by}))
+		{
+			return $user->authorise(
+					'core.edit.own', $componentKey->realcomponent . '.' . $item->id
+				) && $item->{$componentKey->fields->created_by} == $user->id;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if user is allowed to edit item
+	 *
+	 * @param   string  $componentKey  The component properties.
+	 * @param   JTable  $item          Database row from the component.
+	 *
+	 * @return  boolean.
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public static function allowEdit($componentKey = '', $item = null)
+	{
+		$user = JFactory::getUser();
+
+		// Different case for menu items
+		if ($componentKey->realcomponent == 'com_menus')
+		{
+			if (isset($item->menutypeid))
+			{
+				return $user->authorise('core.edit', 'com_menus.menu.' . $item->menutypeid);
+			}
+			else
+			{
+				$table = JTable::getInstance('MenuType');
+				$table->load(array('menutype' => $item->menutype));
+				
+				return $user->authorise('core.edit', 'com_menus.menu.' . $table->id);
+			}
+			
+		}
+
+		return $user->authorise('core.edit', $componentKey->realcomponent . '.' . $item->id) || self::allowEditOwn($componentKey, $item);
+	}
+
+	/**
+	 * Check if user is allowed to create item
+	 *
+	 * @param   string  $componentKey  The component properties.
+	 *
+	 * @return  boolean.
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public static function allowCreate($componentKey = '')
+	{
+		$user = JFactory::getUser();
+		
+		return $user->authorise('core.create', $componentKey->realcomponent);
```

Comment:
Ok, I remember now, if core.create is disabled for menus, user won't be able to create a new item, i.e., when choosing target, it won't be clickable if it's new. Regarding specific permissions for menu parents, if you try to pick a parent, the default ACL from the component itself won't allow you, are you thinking on an extra case where this might break?